src/backend/web/handlers/team_admin.py

```diff
@@ -0,0 +1,252 @@
+
+from backend.common.auth import current_user
+from backend.common.consts.suggestion_state import SuggestionState
+from backend.common.manipulators.media_manipulator import MediaManipulator
+from backend.common.manipulators.robot_manipulator import RobotManipulator
+from backend.common.models.media import Media
+from backend.common.models.robot import Robot
+from backend.common.models.suggestion import Suggestion
+from backend.common.models.team import Team
+from backend.common.models.team_admin_access import TeamAdminAccess
+from backend.common.queries.media_query import TeamSocialMediaQuery
+from backend.web.decorators import require_login
+from backend.web.profiled_render import render_template
+
+blueprint = Blueprint("team_admin", __name__, url_prefix="/")
+
+ALLOWED_SUGGESTION_TYPES = ["media", "social-media", "robot"]
+SUGGESTION_NAMES = {
+    "media": "Media",
+    "social-media": "Social Media",
+    "robot": "Robot CAD",
+}
+SUGGESTION_REVIEW_URL = {
+    "media": "/suggest/team/media/review",
+    "social-media": "/suggest/team/social/review",
+    "robot": "/suggest/cad/review",
+}
+
+
+@blueprint.route("/mod", methods=["GET"])
+@require_login
+def team_mod():
+    user = none_throws(current_user()).account_key
+
+    now = datetime.now()
+    existing_access = TeamAdminAccess.query(
+        TeamAdminAccess.account == user, TeamAdminAccess.expiration > now
+    ).fetch()
+
+    # If the current user is an admin, allow them to view this page for any
+    # team/year combination
+    forced_team = request.args.get("team")
+    forced_year = request.args.get("year")
+    if current_user().is_admin and forced_team and forced_year:
+        existing_access.append(
+            TeamAdminAccess(
+                team_number=int(forced_team),
+                year=int(forced_year),
+            )
+        )
+
+    team_keys = [
+        ndb.Key(Team, "frc{}".format(access.team_number)) for access in existing_access
+    ]
+    if not team_keys:
+        return redirect(url_for(".team_admin_redeem"))
+
+    years = set([access.year for access in existing_access])
+    teams_future = ndb.get_multi_async(team_keys)
+    robot_keys = [
+        ndb.Key(Robot, Robot.render_key_name(team.id(), now.year)) for team in team_keys
+    ]
+    robots_future = ndb.get_multi_async(robot_keys)
+    social_media_futures = [
+        TeamSocialMediaQuery(team_key.id()).fetch_async() for team_key in team_keys
+    ]
+    team_medias_future = Media.query(
+        Media.references.IN(team_keys), Media.year.IN(years)
+    ).fetch_async(50)
+    suggestions_future = (
+        Suggestion.query(Suggestion.review_state == SuggestionState.REVIEW_PENDING)
+        .filter(Suggestion.target_model.IN(ALLOWED_SUGGESTION_TYPES))
+        .filter(Suggestion.target_key.IN([k.id() for k in team_keys]))
+        .fetch_async(limit=50)
+    )
+
+    team_num_to_team = {
+        team.get_result().team_number: team.get_result() for team in teams_future
```

Comment:
Is there some way to DRY this so we don't have to write the `team.get_result()` twice?